solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,158 @@
+     * @return стрим
+     */
+    public Streams<T> filter(@NonNull Predicate<T> object) {
+        Function<List<T>, List<T>> operation = (toFilter) -> {
+            List<T> result = new ArrayList<>();
+            for (T it : toFilter) {
+                if (object.test(it)) {
+                    result.add(it);
+                }
+            }
+            return result;
+        };
+        return new Streams<>(operation.apply(list));
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param function - правило траснформации элементов.
+     * @return стрим
+     */
+    public <R> Streams<R> transform(@NonNull Function<T, R> function) {
+        Function<List<T>, List<R>> operation = (toTransform) -> {
+            List<R> result = new ArrayList<>();
+            for (T it : toTransform) {
+                result.add(function.apply(it));
+            }
+            return result;
+        };
+        return new Streams<>(operation.apply(list));
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
+     * Аналог Stream#sorted
+     * <p>
+     * Intermediate операция.
+     *
+     * @param comparator - правило сортировки элементов.
+     * @return стрим
+     */
+    public Streams<T> sorted(@NonNull Comparator<? super T> comparator) {
+        Function<List<T>, List<T>> operation = (toSort) -> {
+            toSort.sort(comparator);
+            return toSort;
+        };
+        return new Streams<>(operation.apply(list));
+    }
+
+    /**
+     * Преобразование стрима в Map. Необходимо подобрать нужные интерфейсы для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Map используя keyMapper и valueMapper.
+     * <p>
+     * Terminate операция.
+     *
+     * @param keyMapper   - правило создания ключа.
+     * @param valueMapper - правило создания значения.
+     * @return Map, собранная по правилам keyMapper и valueMapper
+     */
+    public <K, V> Map<K, V> toMap(@NonNull Function<T , K> keyMapper,@NonNull  Function<T , V> valueMapper) {
+        List<T> resultList = new ArrayList<>(list);
+        Map<K, V> map = new HashMap<>();
+        for (T it : resultList) {
+            map.put(keyMapper.apply(it), valueMapper.apply(it));
+        }
+        return map;
+    }
+
+    /**
+     * Преобразование стрима в Set.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Set. Обратите внимание, что метод sorted должен учитываться
+     * при вызове этого метода.
+     * <p>
```

Comment:
&lt;p&gt; tag should be preceded with an empty line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-325757449>*